logstreams/src/main/java/io/zeebe/logstreams/impl/log/Listener.java

```diff
@@ -27,13 +28,17 @@ public void onWrite(final long address) {
 
   @Override
   public void onWriteError(final Throwable error) {
-    LogStorageAppender.LOG.error(
-        "Failed to append block with last event position {}.", highestPosition, error);
-    if (error instanceof NoSuchElementException) {
+    if (error instanceof NoSuchElementException || error instanceof NoLeader) {
       // Not a failure. It is probably during transition to follower.
+      LogStorageAppender.LOG.warn(
+          "Failed to append block with last event position {}. This can happen during a leader change.",
+          highestPosition,
+          error);
```

Comment:
Since this case is not a failure, do we want to reduce the log level to `DEBUG` instead?
Or, is it useful to see these log statements on `WARN`?

Reply:
Agree. DEBUG would be better. I changed it.